src/Hangfire.Core/Dashboard/Content/js/hangfire.js

```diff
@@ -457,8 +459,177 @@
                     e.preventDefault();
                 });
 
+
+                $(this).on('click', '.js-jobs-filter-command', function (e) {
+                    
+                    url = document.location.search;
+
+                    var filterValueString = $("#filterValueString").val();
+                    var filterStartDate = $("#startDate").datepick({ dateFormat: 'dd-mm-yyyy' }).val();
+                    var filterEndDate = $("#endDate").datepick({ dateFormat: 'dd-mm-yyyy' }).val();
+
+                    var addedFilterString = prepFilterStringParameter(filterValueString);    
+                    
+                    var addedDateFilterStrings = prepFilterDateParameters(filterStartDate, filterEndDate);
+                    
+                    
+                    if (addedDateFilterStrings || addedFilterString) {                                  
+                        redirectToFirstPage();                        
+                    }                    
+                    document.location.search = url;
+                                      
+                })
+
+
+                var prepFilterStringParameter = function (filterString) {
+                    
+                    var result = false;
+                    if (url == '' && filterString != '') {                        
+                        url = '?' + "filterString=" + filterString;                       
+                        result = true;
+                    }
+                    else {
+                        var parameters = url.substr(1).split('&');
```

Comment:
This section of code is used a few times, Try putting it into a function so it can be reused in the code